Add snackbar on selection of inappropriate choice

erikatan opened the pull request:
If a user selects a bad decision, a snackbar will show saying "Are you sure this is the right decision to take?" and it will allow the user to undo the decision. Another snackbar will say that the session is "restored" and the user can now pick another response.

After clicking a bad response:
<img width="312" alt="screen shot 2017-01-02 at 10 10 33 am" src="https://cloud.githubusercontent.com/assets/24324656/21591624/dd073ffe-d0d3-11e6-8d02-41fc29804dcb.png">

After clicking undo:
<img width="247" alt="screen shot 2017-01-02 at 10 10 53 am" src="https://cloud.githubusercontent.com/assets/24324656/21591627/e2ae3cf0-d0d3-11e6-8bfc-8254e1bbfb3a.png">

anubhakushwaha commented on PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java:
```diff
@@ -135,6 +154,7 @@ public void onItemClick(AdapterView<?> arg0, View view,
                         }
                     }
                 });
+
```
From line 149-156 is code repeatition,please remove

anubhakushwaha commented on PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java:
```diff
@@ -112,21 +113,39 @@ public void onCreate(Bundle savedInstanceState) {
+                                   SessionHistory.currSessionID = scene.getNextScenarioID();
+                                   if (SessionHistory.currSessionID == -1) {
+                                       // Check to make sure all scenes are completed
+                                       SessionHistory.currSessionID == 1;
```
Correct this

anubhakushwaha commented on PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java:
```diff
@@ -112,21 +113,39 @@ public void onCreate(Bundle savedInstanceState) {
-
+                                            final int position, long id) {
+                        if (answers.get(position).getPoints() == 0 &&
+                            !(answers.get(position).getAnswerDescription().equals(R.string.returns_home))) {
```
try matches instead of equals since on running your code it's showing a snackbar even on selection "Returns Home"

anubhakushwaha commented:
@erikatan I have also commented on the GCI web-app about some other changes,please follow up :)

erikatan commented:
Changes made:
1. Made count variable to show snackbar once per bad decision
2. Changed equals() to matches(). "Returns Home" is now fixed
3. Game will now continue successfully if the same bad decision is chosen again on purpose.

chhavip commented:
Has this task been approved in GCI? @anubhakushwaha @erikatan ?

anubhakushwaha commented:
@chhavip Yes
